llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -11237,6 +11237,12 @@ void SelectionDAG::salvageDebugInfo(SDNode &N) {
   if (!N.getHasDebugValue())
     return;
 
+  auto handle = [](SDNode *Node, unsigned ResNo) {
```

Comment:
I think we could improve this name - `GetLocationOperand` or something like that?

nit: lambda naming tends to follow normal variable naming rules (capitalized).